### ISSUE TITLE
[ads] Consider custom client and server error responses as error pages

### ISSUE THREAD

**Issue by btlechowski** [brave/brave-browser#40520]
### Description

Follow up to: https://github.com/brave/brave-browser/issues/39706

This is a regression introduced in 1.69.x. Not reproducible in 1.68.x.

### Steps to reproduce

1. Set targetUrl of an ad to a 404 page
2. View the ad
3. Land the ad
4. Check `landed` confirmation for `"httpResponseStatus": "errorPage"`

### Actual result

`"httpResponseStatus": "errorPage"` is not included in the request

### Expected result

`"httpResponseStatus": "errorPage"` is included in the request

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave | 1.69.139 Chromium: 127.0.6533.100 (Official Build) beta (64-bit)
-- | --
Revision | 0da988816510c06ff835a14b500d5ab861e18d6e
OS | Linux

cc @tmancey @iambrianfung @aseren

**Comment by kjozwiak** [comment]
The above requires `1.69.152` or higher for `1.69.x` verification 👍

**Comment by btlechowski** [comment]
Verified with

Brave | 1.69.153 Chromium: 128.0.6613.85 (Official Build) (64-bit)
-- | --
Revision | a69e96756922f88cbce3482971d70cfd1ee6f872
OS | Linux


#### 404

<details>

![image](https://github.com/user-attachments/assets/dcd45f76-b33e-48c1-8b6d-132e8d8c43a2)


Verified `"httpResponseStatus": "errorPage"` is included in the confirmation

```
[31770:31770:0822/152430.264770:VERBOSE1:redeem_reward_confirmation.cc(75)] Redeem reward confirmation
[31770:31770:0822/152430.268661:VERBOSE1:redeem_reward_confirmation.cc(90)] Create reward confirmation
[31770:31770:0822/152430.268709:VERBOSE6:redeem_reward_confirmation.cc(95)] URL Request:
  URL: https://anonymous.ads.bravesoftware.com/v3/confirmation/204c6e39-7213-4bf9-b6f6-632042c9b6bc/eyJzaWduYXR1cmUiOiJYb0tqT3AvaDJ5dTBhcy82c3B4OEh4RlozcE40SmJNYmo2YnF5K0cvamtoTURweGtTeUV1U3hJaXJUZXRtdlJuUmxxejhDQVVNNmpyWnZHbXBPeUEvZz09IiwidCI6InFJek5Dbm50cklocWY2bnNYV0Mwb01xajBobndMSjJYc25CTU5DQjdEWUd6ZWgxRi83WUFvWmJzQU1rTkhySndlU0EyS1BVbGNUSG5aUzZpT1V4RWN3PT0ifQ==
  Content: {"blindedPaymentTokens":["oFqrqWjGoADsToZGpUcqbLEcwFkK5+MmGYz3udsFKGE="],"buildChannel":"release","catalog":[{"id":"8fdb249064646ae0d63c53a6ed9f0f0747b46da9"}],"countryCode":"US","createdAtTimestamp":"2024-08-22T13:00:00.000Z","creativeInstanceId":"e4958d00-e35c-4134-a408-1fbcf274d5ae","httpResponseStatus":"errorPage","platform":"linux","publicKey":"hLaC3RQw5++hx8zaT+47CA8r8cOlAm+ygeky/ci6T1s=","rotatingHash":"8WKlwmX3sJuQkjEwfFchf1GqORNWm5wCMDibdvlehTc=","segment":"untargeted","studies":[],"systemTimestamp":"2024-08-22T13:00:00.000Z","transactionId":"204c6e39-7213-4bf9-b6f6-632042c9b6bc","type":"landed","versionNumber":"128.0.6613.85"}

```


</details>



#### 500

<details>

![image](https://github.com/user-attachments/assets/d26b0ee9-fcaf-404d-85a3-3dc84e5bb98c)


Verified `"httpResponseStatus": "errorPage"` is included in the confirmation

```
[30429:30429:0822/151757.942780:VERBOSE1:redeem_reward_confirmation.cc(75)] Redeem reward confirmation
[30429:30429:0822/151757.944917:VERBOSE1:redeem_reward_confirmation.cc(90)] Create reward confirmation
[30429:30429:0822/151757.944963:VERBOSE6:redeem_reward_confirmation.cc(95)] URL Request:
  URL: https://anonymous.ads.bravesoftware.com/v3/confirmation/ebeb9775-79c8-4fb2-a45f-9a6151f17790/eyJzaWduYXR1cmUiOiI4cVM5V2tPTjZMVnJLamFpRVlkVFlwOWJXSzRMSVhTSlBFZXhsTG8rWnhMMDFXYzNPWlZjK0dQNlkwMjZ3Nm1vZlV2dFZVbkppby9tQW9KV0tCemZxQT09IiwidCI6IjdwVmhPNkd4emhuZE5tOUw1RlZ1dEF2MFE1WTdMMVgwa1hNeFZrblBrT1lTMXJwUlVqZE9uVHVYUTlsejhkaGlVWmRWbXAyVUxsa3RvdytURGtobW13PT0ifQ==
  Content: {"blindedPaymentTokens":["YtWO4aCv3Kt/wNoDHxLQBES1vhlbF1gohg/KWnaCn2I="],"buildChannel":"release","catalog":[{"id":"8fdb249064646ae0d63c53a6ed9f0f0747b46da9"}],"countryCode":"US","createdAtTimestamp":"2024-08-22T13:00:00.000Z","creativeInstanceId":"22110568-4125-4b84-b97a-27ef45c5c07c","httpResponseStatus":"errorPage","platform":"linux","publicKey":"hLaC3RQw5++hx8zaT+47CA8r8cOlAm+ygeky/ci6T1s=","rotatingHash":"2Exm72rUgqtWPELMcidvR+ZlHP3sY+RriGxTxbXWa50=","segment":"untargeted","studies":[],"systemTimestamp":"2024-08-22T13:00:00.000Z","transactionId":"ebeb9775-79c8-4fb2-a45f-9a6151f17790","type":"landed","versionNumber":"128.0.6613.85"}

```


</details>

**Comment by btlechowski** [comment]
Verification passed on Brave v1.69.153 on Lenovo TB-8506FS (Android 11.0)

#### 404

<details>

![image](https://github.com/user-attachments/assets/eb58d878-ff89-4743-a7b9-03d31ff60779)


Verified `"httpResponseStatus": "errorPage"` is included in the confirmation

```
17:18:50.355  V  [VERBOSE1:redeem_reward_confirmation.cc(75)] Redeem reward confirmation
17:18:50.360  V  [VERBOSE1:redeem_reward_confirmation.cc(90)] Create reward confirmation
17:18:50.361  V  [VERBOSE6:redeem_reward_confirmation.cc(95)] URL Request:
                   URL: https://anonymous.ads.bravesoftware.com/v3/confirmation/64df7e7b-6419-4b35-a5bd-df4e0ef745ec/eyJzaWduYXR1cmUiOiJtYnNGNVJ6VGcwVHc5dEI1RVVDeCs4TEtlbStpUTFiUUNyaDIxSnJ2R3BncDNVTDFQeHBwVXVyTkQ4YUtoamtmS3pmcjZPcWMzeHF6a0gzcW5PdGNFQT09IiwidCI6IlhncGZlQzFzdEdhZU4xdndYS0FhZFpFcTZqaHFyKzFCOE9ER2x2dFRuTkkzT2RMTFRmV0VYcXFVczNPZjBPOTdRWldHTjFJY1IyQkxwQ2U4dHBXNDZnPT0ifQ==
                   Content: {"blindedPaymentTokens":["/GVZGFqBef/loCZkdChl52r9MWOPBMBQc8VcfbtWBxo="],"buildChannel":"release","catalog":[{"id":"8fdb249064646ae0d63c53a6ed9f0f0747b46da9"}],"countryCode":"US","createdAtTimestamp":"2024-08-22T15:00:00.000Z","creativeInstanceId":"1d87cb97-3f55-4a3d-bce1-1c841f39b6f8","httpResponseStatus":"errorPage","platform":"android","publicKey":"hLaC3RQw5++hx8zaT+47CA8r8cOlAm+ygeky/ci6T1s=","rotatingHash":"lZiG+ygtGvNpz5vp/RpmHv+vLnQFBteDPaI2C9HCl5g=","segment":"untargeted","studies":[],"systemTimestamp":"2024-08-22T15:00:00.000Z","transactionId":"64df7e7b-6419-4b35-a5bd-df4e0ef745ec","type":"landed","versionNumber":"128.0.6613.85"}
                   Content Type: application/json
                   Method: kPost

```




</details>



#### 500

<details>

![image](https://github.com/user-attachments/assets/1dd739d8-c74d-4b02-a9a4-85205517e2a3)


Verified `"httpResponseStatus": "errorPage"` is included in the confirmation

```
17:27:28.412  V  [VERBOSE1:redeem_reward_confirmation.cc(75)] Redeem reward confirmation
17:27:28.420  V  [VERBOSE1:redeem_reward_confirmation.cc(90)] Create reward confirmation
17:27:28.421  V  [VERBOSE6:redeem_reward_confirmation.cc(95)] URL Request:
                   URL: https://anonymous.ads.bravesoftware.com/v3/confirmation/fcd39bbc-45a2-46f9-8d9c-0d298a86e472/eyJzaWduYXR1cmUiOiI2UmpDblRZUzFkNmpaMWpBaVE0VXZSUHA2a3pKeFY5Y0lOa0JaMzJad1dBUEdlMGJ2T0dRRTdscFRWc0NNSU1XMjdkcnpWNGtUa0w2eWtUZzBIbFRSUT09IiwidCI6IjljcENSbHFhUG1zaEZmTTMrMUxMVFZYaWNiWXdrK3dQeHF1Vm1hQWlsZVNqRDcyTUZaM2lMbnRBZE1JWjdKYkx6VXc1dWxlQStQWjg0ckppL2grbTlRPT0ifQ==
                   Content: {"blindedPaymentTokens":["JP33/3oC9wEYUPfhAasL64e1RcEGLVqDwUXO00pjMF0="],"buildChannel":"release","catalog":[{"id":"8fdb249064646ae0d63c53a6ed9f0f0747b46da9"}],"countryCode":"US","createdAtTimestamp":"2024-08-22T15:00:00.000Z","creativeInstanceId":"cbe9f40d-257f-429c-939c-39f3d05a0372","httpResponseStatus":"errorPage","platform":"android","publicKey":"hLaC3RQw5++hx8zaT+47CA8r8cOlAm+ygeky/ci6T1s=","rotatingHash":"uLV0bvDPd91sol5fmZHjDOyLKX50WHflbkSiBFOfoEc=","segment":"untargeted","studies":[],"systemTimestamp":"2024-08-22T15:00:00.000Z","transactionId":"fcd39bbc-45a2-46f9-8d9c-0d298a86e472","type":"landed","versionNumber":"128.0.6613.85"}
                   Content Type: application/json
                   Method: kPost
```


</details>
